lib/Dashboard/MailWidget.php

```diff
@@ -95,4 +124,88 @@ public function load(): void {
 			$this->accountService->findByUserId($this->userId)
 		);
 	}
+
+	/**
+	 * Get widget-specific search filter
+	 * @return string
+	 */
+	abstract public function getSearchFilter(): string;
+
+	/**
+	 * @param string $userId
+	 * @return Message[]
+	 * @throws ClientException
+	 * @throws ServiceException
+	 */
+	protected function getEmails(string $userId): array {
+		$accounts = $this->accountService->findByUserId($userId);
+		$filter = $this->getSearchFilter();
+		$emails = [];
+
+		foreach ($accounts as $account) {
+			$mailboxes = $this->mailManager->getMailboxes($account);
+			/** @var Mailbox[] $mailboxes */
+			$mailboxes = array_filter($mailboxes, function (Mailbox $mailbox) {
+				return $mailbox->isInbox();
+			});
+			foreach ($mailboxes as $mailbox) {
+				$searchResult = $this->mailSearch->findMessages(
+					$account,
+					$mailbox,
+					$filter,
+					null,
+					10
+				);
+				$emails = array_merge($emails, $searchResult);
+			}
+		}
+
+		usort($emails, function (Message $a, Message $b) {
+			return ($a->getSentAt() > $b->getSentAt()) ? -1 : 1;
+		});
+		return $emails;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function getItems(string $userId, ?string $since = null, int $limit = 7): array {
+		$emails = $this->getEmails($userId);
+
+		if ($since !== null) {
```

Comment:
it would pass this down to the `getEmails` call so irrelevant messages are never loaded into memory

Reply:
Is it possible to filter by `send_at`? If so, I didn't get how to do it by looking at `FilterStringParser::parse()`.